homeassistant/components/rki_covid/sensor.py

```diff
@@ -0,0 +1,172 @@
+
+from datetime import datetime
+import logging
+from typing import Any
+
+from homeassistant import config_entries, core
+from .coordinator import RkiCovidDataUpdateCoordinator
+from homeassistant.components.sensor import STATE_CLASS_MEASUREMENT, SensorEntity
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import (
+    ConfigType,
+    DiscoveryInfoType,
+    HomeAssistantType,
+)
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import ATTR_COUNTY, ATTRIBUTION, CONF_DISTRICT_NAME, CONF_DISTRICTS, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSORS = {
+    "count": "mdi:virus",
+    "deaths": "mdi:cross",
+    "recovered": "mdi:bottle-tonic-plus-outline",
+    "week_incidence": "mdi:clipboard-pulse",
+    "cases_per100k": "mdi:home-group",
+    "new_cases": "mdi:shield-bug",
+    "new_deaths": "mdi:shield-cross",
+    "new_recovered": "mdi:shield-sync",
+}
+
+DISTRICT_SENSORS = {
+    "hospitalization_cases_baby": "mdi:baby-face-outline",
+    "hospitalization_incidence_baby": "mdi:baby-face",
+    "hospitalization_cases_children": "mdi:account-child-outline",
+    "hospitalization_incidence_children": "mdi:account-child",
+    "hospitalization_cases_teen": "mdi:face-woman",
+    "hospitalization_incidence_teen": "mdi:face-woman-outline",
+    "hospitalization_cases_grown": "mdi:face-man",
+    "hospitalization_incidence_grown": "mdi:face-man-outline",
+    "hospitalization_cases_senior": "mdi:account-cowboy-hat-outline",
+    "hospitalization_incidence_senior": "mdi:account-cowboy-hat",
+    "hospitalization_cases_old": "mdi:human-white-cane",
+    "hospitalization_incidence_old": "mdi:human-cane",
+}
+
+
+async def async_setup_platform(
+    hass: HomeAssistantType,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the sensor platform."""
+    coordinator = RkiCovidDataUpdateCoordinator(hass)
+    await coordinator.async_config_entry_first_refresh()
+
+    if coordinator is None or coordinator.data is None:
+        raise PlatformNotReady("Data coordinator could not be initialized!")
+
+    districts = config[CONF_DISTRICTS]
+
+    # district sensors
+    sensors = [
+        RKICovidNumbersSensor(coordinator, district[CONF_DISTRICT_NAME], info_type)
+        for info_type in SENSORS
+        for district in districts
+    ]
+    async_add_entities(sensors, update_before_add=True)
+
+    # sensors on state level (additional sensors)
+    for district in districts:
+        if district.startswith("BL"):
+            district_sensors = [
+                RKICovidNumbersSensor(coordinator, district, info_type)
+                for info_type in DISTRICT_SENSORS
+            ]
+            async_add_entities(district_sensors, update_before_add=True)
```

Comment:
Should be removed as it's setup from entry only (config flow)

Reply:
Can you point me what exactly I need to change? In my understanding this integration should use the platform concept since it is containing multiple sensors for a single unit.
Can't figure out what I have to change here.

Reply:
async_setup_platform is used when you load the platform from config.yaml. Here you use a config flow so it will use async_setup_entry